python/pyspark/sql/tests.py

```diff
@@ -3021,6 +3021,17 @@ def test_sort_with_nulls_order(self):
 
 class HiveSparkSubmitTests(SparkSubmitTests):
 
+    @classmethod
+    def setUpClass(cls):
+        import glob
+        from pyspark.find_spark_home import _find_spark_home
+
+        SPARK_HOME = _find_spark_home()
+        filename_pattern = ("sql/hive/target/spark-hive_*-sources.jar")
+        if not glob.glob(os.path.join(SPARK_HOME, filename_pattern)):
+            raise unittest.SkipTest(
```

Comment:
Will this result in the right kind of message, particularly the kind that @HyukjinKwon is checking for in PR #21107?

Reply:
Yea, actually, I think it would be nicer in `setUp` for a better(?) message .. I replaced `unittest.SkipTest` to `self.skipTest` per https://docs.python.org/2/library/unittest.html#unittest.SkipTest and https://docs.python.org/2/library/unittest.html#unittest.TestCase.skipTest

Reply:
I think we should see if @holdenk likes this or not too while we are here.

Reply:
I do like more infromation on skipped tests.

Reply:
Yep. According to the latest convention of #21107, it will be displayed like this.
```
Skipped tests in pyspark.sql.tests with python2.7:
...
    test_hivecontext (pyspark.sql.tests.HiveSparkSubmitTests) ... skipped 'Hive is not available.'
```